### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by andresailer**

### CODE REVIEW

**Comment by gaede** [comment]
Why do you use a global variable here:
https://github.com/iLCSoft/Marlin/pull/8/commits/12e00ec064e14db0ac831050bca756a2202286f0#diff-6c43096941a8c63137f1b9c8718d0c0bR9

File: source/include/marlin/CCProcessor.h [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #include <set>
 #include <fstream>
 
-#define MAX_ERRORS 3
+const int MAX_ERRORS=3;
```
Comment (andresailer) [inline source/include/marlin/CCProcessor.h]: Because a const variable is better than define.

Comment (andresailer) [inline source/include/marlin/CCProcessor.h]: But actually should use constexpr

**Comment by gaede** [comment]
On could argue about that. In any case changing only one out of ten #defines to a const int seems awkward to me as it confuses readers of the code. 
Can you please change back rto #define  ( or consistently replace all defines )

**Comment by andresailer** [comment]
No I really don't think you can argue about the fact that constexpr is better than define. But I will revert because the define was not related to any warning.

**Comment by gaede** [comment]
Yes one can argue about that in certain use cases. For example would your suggested solution:
```c++
constexpr int MAX_ERRORS=3;
```
have introduced a global variable in the global namespace and prevented inclusion of the header file by more than one  *.cc file in the compilation unit. 
Admittedly not a problem in this particular case, but an example where a #define would have been the better choice ( though not necessarily the best, of course).

Anyways, thanks for fixing the warnings - I'll merge the PR.

**Comment by andresailer** [comment]
A *.cc file is a compilation unit, unless you are doing weird things(i.e. including another cc file). One can effectively only include the header file once in a compilation unit, because of the header guards. And even including the header in an additional processor doesn't break the compilation. Now using non-const variables would cause linker errors if the header file is included more than once.

**Comment by gaede** [comment]
I meant of course linking-unit - and it does not make it better that it compiles but crashes the linking step...

**Comment by andresailer** [comment]
But it links with const or constexpr.

**Comment by gaede** [comment]
oops, you are right of course  - seems I got this confused with C...

**Comment by andresailer** [comment]
ReleaseNotes:
* Fix warnings for GCC
* moved tinyXML to separate library

**Pull request merged**